extensions/typescript-language-features/src/languageFeatures/completions.ts

```diff
@@ -536,15 +535,47 @@ class TypeScriptCompletionItemProvider implements vscode.CompletionItemProvider<
 			useFuzzyWordRangeLogic: this.client.apiVersion.lt(API.v390),
 		};
 
+		let includesPackageJsonImport = false;
 		const items: MyCompletionItem[] = [];
 		for (let entry of entries) {
 			if (!shouldExcludeCompletionEntry(entry, completionConfiguration)) {
 				items.push(new MyCompletionItem(position, document, entry, completionContext, metadata));
+				// @ts-expect-error - remove after TS 4.0 protocol update
+				includesPackageJsonImport = !!entry.isPackageJsonImport;
 			}
 		}
+		if (duration !== undefined) {
+			this.logCompletionsTelemetry(duration, response, includesPackageJsonImport);
+		}
 		return new vscode.CompletionList(items, isIncomplete);
 	}
 
+	private logCompletionsTelemetry(
+		duration: number,
+		response: ServerResponse.Response<Proto.CompletionInfoResponse> | undefined,
+		includesPackageJsonImport?: boolean
+	) {
+		/* __GDPR__
+			"completions.execute" : {
+				"duration" : { "classification": "SystemMetadata", "purpose": "FeatureInsight" },
```

Comment:
Note: I changed the classification of these properties at @amcasey’s request.

Reply:
Generally speaking, numbers computed without consuming personal data are "SystemMetadata".